src/sentry/api/endpoints/shared_group_details.py

```diff
@@ -42,9 +45,14 @@ def get(
         except Group.DoesNotExist:
             raise ResourceDoesNotExist
 
+        # Checks if the organization_slug (eventually renamed to organization_id_or_slug) matches the group organization's id or slug
         if organization_slug:
-            if organization_slug != group.organization.slug:
-                raise ResourceDoesNotExist
+            if organization_slug.isnumeric():
```

Comment:
tiny nit: I think it's slightly [safer](https://stackoverflow.com/a/54912545) to use `isdecimal` over `isnumeric`